src/main/java/org/spongepowered/api/item/recipe/RecipeRegistry.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.recipe;
+
+import java.util.Set;
+
+/**
+ * A RecipeRegistry holds all registered recipes for a given game.
+ */
+public interface RecipeRegistry {
+
+    /**
+     * Registers the given Recipe to make it available to craft.
+     *
+     * @param recipe The Recipe to register
+     */
+    void register(Recipe recipe);
+
+    /**
+     * Removes the given Recipe from registration in this registry.
+     *
+     * @param recipe The Recipe to unregister
+     */
+    void remove(Recipe recipe);
```

Comment:
Return `boolean` whether it was removed? (recipe may not have existed in the registry)